builtin/game/falling.lua

```diff
@@ -146,13 +146,28 @@ end
 
 local function drop_attached_node(p)
 	local n = core.get_node(p)
+	local node_meta, meta_fields = core.get_meta(p), nil
+	if node_meta then meta_fields = node_meta:to_table().fields end
 	core.remove_node(p)
 	for _, item in pairs(core.get_node_drops(n, "")) do
 		local pos = {
 			x = p.x + math.random()/2 - 0.25,
 			y = p.y + math.random()/2 - 0.25,
 			z = p.z + math.random()/2 - 0.25,
 		}
+		-- If it drops itself, preserve its metadata.
+		if item == n and meta_fields then
+			local stack, stack_meta = ItemStack({name=n, count=1}), nil
+			if stack then stack_meta = stack:get_meta() end
+			if stack_meta then
+				for k,v in pairs(meta_fields) do
+					if k ~= "description" and k ~= "infotext" and k ~= "formspec" then
+						stack_meta:set_string(k,v )
+					end
+				end
+			end
+			item = stack:to_string()
```

Comment:
should work without

Reply:
I was trying to find the line that said that, could not before writing the code.
Of course, now I can:
// add_item(self, itemstack or itemstring or table or nil) -> itemstack